mmf/datasets/builders/vqacp_v2/database.py

```diff
@@ -0,0 +1,88 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+import copy
+import json
+import os
+
+from mmf.datasets.databases.annotation_database import AnnotationDatabase
+from mmf.datasets.databases.image_database import (
+    ImageDatabase,
+    get_possible_image_paths,
+)
+from mmf.utils.file_io import PathManager
+
+
+class VQACPv2AnnotationDatabase(AnnotationDatabase):
+    def __init__(self, config, path, *args, **kwargs):
+        path = path.split(",")
+        super().__init__(config, path, *args, **kwargs)
+
+    def load_annotation_db(self, path):
+        # Expect two paths, one to questions and one to annotations
+        assert (
+            len(path) == 2
+        ), "VQACPv2 requires 2 paths; one to questions and one to annotations"
+
+        with PathManager.open(path[0]) as f:
+            path_0 = json.load(f)
+        with PathManager.open(path[1]) as f:
+            path_1 = json.load(f)
+
+        if "annotations" in path[0]:
+            annotations = path_0
+            questions = path_1
+        else:
+            annotations = path_1
+            questions = path_0
+
+        # Convert to linear format
+        data = []
+        question_dict = {}
+        for question in questions:
+            question_dict[question["question_id"]] = question["question"]
+
+        for annotation in annotations:
+            annotation["question"] = question_dict[annotation["question_id"]]
+            answers = []
+            for answer in annotation["answers"]:
+                answers.append(answer["answer"])
+            annotation["answers"] = answers
+            data.append(copy.deepcopy(annotation))
+
+        self.data = data
+
+
+class VQACPv2ImageDatabase(ImageDatabase):
+    def __init__(self, config, path, *args, **kwargs):
+        path = path.split(",")
+        super().__init__(config, path, *args, **kwargs)
+
+    def from_path(self, image_path, use_transforms=True):
+        if "train" in self.base_path[0]:
+            train_path = self.base_path[0]
+            val_path = self.base_path[1]
+        else:
+            train_path = self.base_path[1]
+            val_path = self.base_path[0]
+
+        if "train" in image_path:
```

Comment:
It might be easier to adjust this in the dataset itself in function `preprocess_sample_info`. Then you won't need to update this.

Reply:
Thanks for pointing this out, I adjusted it in the `get_image_path` function in the dataset and then removed the VQACPv2ImageDatabase entirely since it was no longer necessary.